FITS roundtrip of complex values inverts real and imaginary parts

### Description

Roundtripping complex columns from a `Table` to FITS (and only FITS) results in the real and imaginary parts being swapped.

### Expected behavior

The roundtrip of complex values to FITS with `Table.write` should work

### How to Reproduce

pip install "astropy>=0.0.dev"

Run pytest -svv test_file.py

Content of test_file.py:

```python
import os
import pytest
import numpy as np
from astropy.table import Table


@pytest.mark.parametrize("fmt", ["fits", "hdf5"])
def test_file_roundtrip_complex(fmt):
    """Round trip complex data to different formats."""

    table = Table({"a": [1 + 4.3j, 2.2 + 6.1j, 3 + 7.2j]})

    fname = f"dummy.{fmt}"
    table.write(fname, format=fmt)

    new_table = Table.read(fname, format=fmt)
    os.unlink(fname)

    for attr in table.colnames:
        assert np.allclose(table[attr], new_table[attr], rtol=1e-5, atol=1e-8)
```

This should work for both HDF5 and FITS in Astropy up to v.7.2 at least, and fail for FITS in astropy 8.0.dev. In the error message, pytest will show that the real and imaginary parts of the input and output are swapped:

```AssertionError: assert False
E            +  where False = <function allclose at 0x103157b70>(<Column name='a' dtype='complex128' length=3>\n  (1+4.3j)\n(2.2+6.1j)\n  (3+7.2j), <Column name='a' dtype='complex128' length=3>\n  (4.3+1j)\n(6.1+2.2j)\n  (7.2+3j), rtol=1e-05, atol=1e-08)
E            +    where <function allclose at 0x103157b70> = np.allclose
```

### Versions

```
platform
--------
platform.platform() = 'macOS-26.5-arm64-arm-64bit-Mach-O'
platform.version() = 'Darwin Kernel Version 25.5.0: Mon Apr 27 20:38:56 PDT 2026; root:xnu-12377.121.6~2/RELEASE_ARM64_T6000'
platform.python_version() = '3.13.12'

packages
--------
astropy              8.0.0b1
numpy                2.4.6
scipy                1.17.1
matplotlib           3.10.8
pandas               --
pyerfa               2.0.1.4
```

--- a/astropy/io/fits/hdu/table.py
+++ b/astropy/io/fits/hdu/table.py
@@ -1528,13 +1528,21 @@ def _as_bigendian_pieces(data):
         # the .str will be '|V<N>' where <N> is the total bytes per element.
         if field_dtype != dtype_big.fields[name][0]:
             offset = field_offset
-            itemsize = field_dtype.base.itemsize
-            # Reorder every element of a possible sub-array.
-            for i in range(math.prod(field_dtype.shape)):
-                indices[offset : offset + itemsize] = indices[
-                    offset : offset + itemsize
+            # For complex types, swap each float component independently
+            # rather than reversing all bytes of the complex number
+            # (which would swap real and imaginary parts).
+            if field_dtype.base.kind == "c":
+                swap_itemsize = field_dtype.base.itemsize // 2
+            else:
+                swap_itemsize = field_dtype.base.itemsize
+            # Reorder every element of a possible sub-array,
+            # and every component of complex numbers.
+            n_swaps = field_dtype.itemsize // swap_itemsize
+            for i in range(n_swaps):
+                indices[offset : offset + swap_itemsize] = indices[
+                    offset : offset + swap_itemsize
                 ][::-1]
-                offset += itemsize
+                offset += swap_itemsize
 
     # Reordering the data by indexing makes copies, so work in pieces.
     data_bytes = data.view(np.ndarray)[..., np.newaxis].view(np.ubyte)

--- a/astropy/io/fits/tests/test_connect.py
+++ b/astropy/io/fits/tests/test_connect.py
@@ -1082,6 +1082,15 @@ def test_info_attributes_with_no_mixins(tmp_path):
     assert t2["col0"].meta["a"] == {"b": "c"}
 
 
+def test_round_trip_complex(tmp_path):
+    """FITS portion of https://github.com/astropy/astropy/issues/19775"""
+    filename = tmp_path / "test.fits"
+    t = Table({"a": [1 + 4.3j, 2.2 + 6.1j, 3 + 7.2j]})
+    t.write(filename, format="fits")
+    t2 = Table.read(filename, format="fits")
+    assert np.allclose(t["a"], t2["a"], rtol=1e-5, atol=1e-8)
+
+
 @pytest.mark.parametrize("method", ["set_cols", "names", "class"])
 def test_round_trip_masked_table_serialize_mask(tmp_path, method):
     """

